wasm/jit/jit_arm64.go

```diff
@@ -1773,36 +1780,211 @@ func (c *arm64Compiler) compileGe(o *wazeroir.OperationGe) error {
 func (c *arm64Compiler) compileStore(o *wazeroir.OperationStore) error {
-	return fmt.Errorf("TODO: unsupported on arm64")
+	var movInst obj.As
+	var targetSizeInBytes int64
+	switch o.Type {
+	case wazeroir.UnsignedTypeI32:
+		movInst = arm64.AMOVW
+		targetSizeInBytes = 32 / 8
+	case wazeroir.UnsignedTypeI64:
+		movInst = arm64.AMOVD
+		targetSizeInBytes = 64 / 8
+	case wazeroir.UnsignedTypeF32:
+		movInst = arm64.AFMOVS
+		targetSizeInBytes = 32 / 8
+	case wazeroir.UnsignedTypeF64:
+		movInst = arm64.AFMOVD
+		targetSizeInBytes = 64 / 8
+	}
+	return c.compileStoreImpl(o.Arg.Offset, movInst, targetSizeInBytes)
 }
 
+// compileStore8 implements compiler.compileStore8 for the amd64 architecture.
 func (c *arm64Compiler) compileStore8(o *wazeroir.OperationStore8) error {
-	return fmt.Errorf("TODO: unsupported on arm64")
+	return c.compileStoreImpl(o.Arg.Offset, arm64.AMOVB, 1)
 }
 
+// compileStore16 implements compiler.compileStore16 for the amd64 architecture.
 func (c *arm64Compiler) compileStore16(o *wazeroir.OperationStore16) error {
-	return fmt.Errorf("TODO: unsupported on arm64")
+	return c.compileStoreImpl(o.Arg.Offset, arm64.AMOVH, 16/8)
 }
 
+// compileStore32 implements compiler.compileStore32 for the amd64 architecture.
 func (c *arm64Compiler) compileStore32(o *wazeroir.OperationStore32) error {
-	return fmt.Errorf("TODO: unsupported on arm64")
+	return c.compileStoreImpl(o.Arg.Offset, arm64.AMOVW, 32/8)
+}
+
+// compileStoreImpl implements compleStore* variants for arm64 architecture.
+func (c *arm64Compiler) compileStoreImpl(offsetArg uint32, storeInst obj.As, targetSizeInBytes int64) error {
+	val, err := c.popValueOnRegister()
+	if err != nil {
+		return err
+	}
+	// Mark temporarily used as compileMemoryAccessOffsetSetup might try allocating register.
+	c.markRegisterUsed(val.register)
+
+	offsetReg, err := c.compileMemoryAccessOffsetSetup(offsetArg, targetSizeInBytes)
+	if err != nil {
+		return err
+	}
+
+	// "[reservedRegisterForMemory + offsetReg] = val.register"
+	// In other words, "memory.Buffer[offset: offset+targetSizeInBytes] = val.register"
+	c.compileRegisterToMemoryWithRegisterOffsetInstruction(
+		storeInst, val.register,
+		reservedRegisterForMemory, offsetReg,
+	)
+
+	c.markRegisterUnused(val.register)
+	return nil
+}
+
+// compileMemoryAccessOffsetSetup pops the top value from the stack (called "base"), stores "base + offsetArg + targetSizeInBytes"
+// into a register, and returns the stored register. We call the result "offset" because we access the memory
+// as memory.Buffer[offset: offset+targetSizeInBytes].
+//
+// Note: this also emits the instructions to check the out of bounds memory access.
+// In other words, if the offset+targetSizeInBytes exceeds the memory size, the code exits with jitCallStatusCodeMemoryOutOfBounds status.
+func (c *arm64Compiler) compileMemoryAccessOffsetSetup(offsetArg uint32, targetSizeInBytes int64) (offsetRegister int16, err error) {
```

Comment:
uint32 or less. ack when adding you may want to cast, but it is distracting here as I don't think the value will be more than single or double digits right?

```suggestion
func (c *arm64Compiler) compileMemoryAccessOffsetSetup(offsetArg uint32, targetSizeInBytes uint32) (offsetRegister int16, err error) {
```